src/pkg/backup/details/details.go

```diff
@@ -313,7 +313,19 @@ func (d *Details) add(
 			filename = info.SharePoint.ItemName
 		}
 
-		entry.ShortRef = p.ToBuilder().Append(filename).ShortRef()
+		// Make the new path be all of the display names and then the M365 item ID.
+		// This ensures the path will be unique, thus ensuring the ShortRef will be
+		// unique.
+		//
+		// If we just appended the file display name to the path then it's possible
+		// the user could have a folder in the parent directory with a display name
+		// equal to the M365 ID of this file and have a file in the folder with a
+		// display name the same as this file's. That would cause a ShortRef
+		// collision because the generated paths for the file in the theoretical
+		// folder and this item would be the same.
+		elements := p.Elements()
+		elements = append(append(elements[:len(elements)-1], filename), p.Item())
```

Comment:
double append not needed
```suggestion
		elements = append(elements[:len(elements)-1], filename, p.Item())
```